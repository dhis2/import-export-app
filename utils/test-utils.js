import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
