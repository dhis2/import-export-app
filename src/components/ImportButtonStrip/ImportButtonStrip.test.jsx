import React from 'react'
import { render } from 'test-utils'

import { ImportButtonStrip } from '../index.js'

const props = {
    dataTest: 'import-strip',
    dryRunDataTest: 'import-strip-dry-run-btn',
    importDataTest: 'import-strip-import-btn',
    form: {},
    onImport: () => 1,
}

const r = <ImportButtonStrip {...props} />

it('matches snapshot', () => {
    const { asFragment } = render(r)
    expect(asFragment()).toMatchSnapshot()
})

it('shows dry-run and import buttons', () => {
    const { getByDataTest } = render(r)
    const strip = getByDataTest(`${props.dataTest}-button-strip`)
    const dryRunBtn = getByDataTest(props.dryRunDataTest)
    const importBtn = getByDataTest(props.importDataTest)
    expect(strip).toBeInTheDocument()
    expect(strip).toContainElement(dryRunBtn)
    expect(strip).toContainElement(importBtn)
})

it('shows help text', () => {
    const { getByDataTest } = render(r)
    const help = getByDataTest(`${props.dataTest}-help`)
    expect(help).toBeInTheDocument()
})
