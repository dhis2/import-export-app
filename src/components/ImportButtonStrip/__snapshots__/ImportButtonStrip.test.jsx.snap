// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="import-strip"
  >
    <div
      class="jsx-34417531 start"
      data-test="import-strip-button-strip"
    >
      <div
        class="jsx-34417531 box"
      >
        <button
          class="jsx-1796590446 dryRun primary"
          data-test="import-strip-dry-run-btn"
          type="submit"
        >
          Start dry run
        </button>
      </div>
      <div
        class="jsx-34417531 box"
      >
        <button
          class="jsx-1796590446 secondary"
          data-test="import-strip-import-btn"
          type="submit"
        >
          Start import
        </button>
      </div>
    </div>
    <p
      class="jsx-4265152438 "
      data-test="import-strip-help"
    >
      A dry run tests the import settings without importing any data
    </p>
  </div>
</DocumentFragment>
`;
